### ISSUE TITLE
'freeze-mods' param for finetuning does not handle 'space' well

### ISSUE THREAD

**Issue by vinitunni** [espnet/espnet#4580]
**Describe the bug**
ESPNET1

Handling of 'spaces' in freeze-mods param is unintuitive and could lead to erroneous results. My experiments were run on a transducer model and I am not sure if the same problem persists for other models.

if we define the parameter as
`freeze-mods: 'dec.,encoder.`, both encoder and decoder are frozen (requires_grad=False)
However, if we define it as  
`freeze-mods: 'dec., encoder.`,  only decoder is frozen (requires_grad=False).
This is probably due to how the string is handled during argparse in espnet/bin/asr.py
`    parser.add_argument(
        "--freeze-mods",
        default=None,
        type=lambda s: [str(mod) for mod in s.split(",") if s != ""],
        help="List of modules to freeze, separated by a comma.",
    )
`
Should get fixed if the lambda function is changed to 
`        type=lambda s: [str(mod).strip() for mod in s.split(",") if s != ""],`

**Comment by b-flo** [comment]
`enc-init-mods`, `dec-init-mods`, `freeze-mods` are defined like that in ESPnet1, for all tasks. It's not a bug in regards to the documentation (we say "separated by a comma" only) but I thought it we already discussed the space case and fixed it.

@sw005320 Should we fix this for ESPnet1?

**Comment by vinitunni** [comment]
I agree the documentation mentions only comma. But I feel it would be more intuitive if the spaces were stripped.
